### PR TITLE
Remove Instances of `first-of-type` selectors

### PR DESCRIPTION
**Pull request by echen-adobe**
## Summary

Briefly describe the features or fixes introduced in this PR.

Replaces fragile CSS selectors such as `:first-of-type` in blocks to prevent certain edge cases caused by MEP overwriting content.

---

## Jira Ticket

Resolves: https://jira.corp.adobe.com/browse/MWPW-184504

---

## Test URLs

| Env | URL |
|-------------|-----|
| **Before**  | https://main--da-express-milo--adobecom.aem.page/express/ |
| **After**   | https://mep-audit--da-express-milo--adobecom.aem.page/docs/library/kitchen-sink/ax-columns |
| **After**   | https://mep-audit--da-express-milo--adobecom.aem.page/docs/library/kitchen-sink/template-x |
| **After**   | https://mep-audit--da-express-milo--adobecom.aem.page/docs/library/kitchen-sink/pricing-cards-v2 |
---

## Verification Steps

- Steps to reproduce the issue or view the new feature.
- What to expect **before** and **after** the change.

---

## Potential Regressions

*AX Columns*

https://mep-audit--da-express-milo--adobecom.aem.page/express/photos

https://mep-audit--da-express-milo--adobecom.aem.page/express/spotlight/switch-now

https://mep-audit--da-express-milo--adobecom.aem.page/express/fragments/max-campaign

https://mep-audit--da-express-milo--adobecom.aem.page/express/fragments/max-entitled-campaign

https://mep-audit--da-express-milo--adobecom.aem.page/express/create/background

https://mep-audit--da-express-milo--adobecom.aem.page/express/create/banner

https://mep-audit--da-express-milo--adobecom.aem.page/express/create/bingo-card

https://mep-audit--da-express-milo--adobecom.aem.page/express/create/brochure

https://mep-audit--da-express-milo--adobecom.aem.page/express/create/business-card

https://mep-audit--da-express-milo--adobecom.aem.page/express/create/video/vlog

https://mep-audit--da-express-milo--adobecom.aem.page/express/create/story/whatsapp-status

https://mep-audit--da-express-milo--adobecom.aem.page/express/create/thumbnail/youtube

https://mep-audit--da-express-milo--adobecom.aem.page/express/create/story/facebook

https://mep-audit--da-express-milo--adobecom.aem.page/express/create/story/instagram

*Template-X-Carousel*

https://mep-audit--da-express-milo--adobecom.aem.page/express/create/advertisement

https://mep-audit--da-express-milo--adobecom.aem.page/express/create/background

https://mep-audit--da-express-milo--adobecom.aem.page/express/create/bingo-card

https://mep-audit--da-express-milo--adobecom.aem.page/express/create/banner

https://mep-audit--da-express-milo--adobecom.aem.page/express/create/brochure

*Template-X*

https://mep-audit--da-express-milo--adobecom.aem.page/express/create

https://mep-audit--da-express-milo--adobecom.aem.page/express/ambassadors

https://mep-audit--da-express-milo--adobecom.aem.page/express/data-visualization

https://mep-audit--da-express-milo--adobecom.aem.page/express/graphic-design

https://mep-audit--da-express-milo--adobecom.aem.page/express/print

https://mep-audit--da-express-milo--adobecom.aem.page/express/social-media

https://mep-audit--da-express-milo--adobecom.aem.page/express/templates/

https://mep-audit--da-express-milo--adobecom.aem.page/express/solutions/blog

https://mep-audit--da-express-milo--adobecom.aem.page/express/solutions/streaming

---

## Additional Notes

(If applicable) Add context, related PRs, or known issues here.

### CODE REVIEW

File: express/code/blocks/ax-columns/ax-columns.js [hunk 1]
```diff
@@ -589,22 +594,13 @@ export default async function decorate(block) {
     );
   }
 
-  if (document.querySelector('main > div > div') === block) {
-    const logo = injectLogo(block);
-    if (logo) {
-      block.querySelector('.column')?.prepend(logo);
-    }
-  }
+  const sections = Array.from(document.querySelectorAll('main > div > div ')).filter((section) => section.dataset.manifestId === undefined);
```
Comment (yeibercano) [inline express/code/blocks/ax-columns/ax-columns.js]:
```suggestion
  const sections = Array.from(document.querySelectorAll('main > div > div')).filter((section) => section.dataset.manifestId === undefined);
```

File: express/code/blocks/template-x/template-x.js [hunk 2]
```diff
@@ -313,7 +319,9 @@ function adjustPlaceholderDimensions(block, props, tmplt, option) {
 }
 
 function adjustTemplateDimensions(block, props, tmplt, isPlaceholder) {
-  const overlayCell = tmplt.querySelector(':scope > div:last-of-type');
+  const divs = tmplt.querySelectorAll(':scope > div');
+  const overlayCell = divs[2];
```
Comment (yeibercano) [inline express/code/blocks/template-x/template-x.js:373]: Are we always accounting there would be three divs? would it be safer to get the last one instead: `const overlayCell = divs[divs.length - 1];`

**Comment by fullcolorcoder** [comment]
Can you add a more reliable list of regression testing links here?

**Review by maxn-adobe — APPROVED** [review]

File: express/code/blocks/template-x/template-x.js (hunk 2 above)
Comment (echen-adobe) [reply express/code/blocks/template-x/template-x.js:373]: Its being called from this part.

`  if (tmplt.children.length === 3) {
      // look for options in last cell
      adjustTemplateDimensions(block, props, tmplt, isPlaceholder);
    }`

Comment (yeibercano) [reply express/code/blocks/template-x/template-x.js:373]: Right, it seems all is very tightly coupled from before.  Lets process: ideally we would want to return early to avoid branching in so many levels with if.

Thank you

**Review by yeibercano — APPROVED** [review]

**Comment by echen-adobe** [comment]
> Can you add a more reliable list of regression testing links here?

Updated just now.

**Review by fullcolorcoder — APPROVED** [review]
